### ISSUE TITLE
Code coverage not appearing in Qube

### ISSUE THREAD

**Issue by sgtsquiggs** [Backelite/sonar-swift#112]
Coverage is being generated properly by slather. I get a few warnings from sonar-scanner, and the project in SQ does not report any code coverage stats.

```
15:03:49.379 INFO: Sensor org.sonar.plugins.swift.coverage.SwiftCoberturaSensor [backelitesonarswiftplugin]
15:03:49.379 INFO: Processing coverage report /Users/mcrenshaw/Projects/Prototype/reports/cobertura.xml
15:03:49.402 WARN: Coverage measure for metric 'lines_to_cover' should not be saved directly by a Sensor. Plugin should be updated to use SensorContext::newCoverage instead.
15:03:49.402 WARN: Coverage measure for metric 'uncovered_lines' should not be saved directly by a Sensor. Plugin should be updated to use SensorContext::newCoverage instead.
15:03:49.402 WARN: Coverage measure for metric 'coverage_line_hits_data' should not be saved directly by a Sensor. Plugin should be updated to use SensorContext::newCoverage instead.
15:03:49.409 INFO: Sensor org.sonar.plugins.swift.coverage.SwiftCoberturaSensor [backelitesonarswiftplugin] (done) | time=30ms
```

**Comment by ButkiewiczP** [comment]
Experiencing the same issue after updating from 5.5 to 6.5.

**Comment by sgtsquiggs** [comment]
I got this working. Instead of using the run_sonar_scanner.sh script, I am now using alternate fastlane setup.

**Comment by singh88** [comment]
I have found the solution to this problem. I was using Fastlane configuration. After running this I found that under reports directory coverage reports was getting generated with name cobertura.xml. So, I have modified the Fastlane configuration script to include a step to rename cobertura.xml to coverage-swift.xml (this string is referenced by sonar-swift code base [here](https://github.com/Backelite/sonar-swift/blob/a404d5f64ae74dcf3bcad42c95a3df083c870fa6/sonar-swift-plugin/src/main/java/org/sonar/plugins/swift/coverage/SwiftCoberturaSensor.java#L39)) and that solved my problem.
Here is the modified fastlane lane.
```
lane :metrics do
    scan(scheme: "[SCHEME]", code_coverage: true, derived_data_path: "./DerivedData", output_directory: "./sonar-reports")
    slather(cobertura_xml: true, jenkins: true, scheme: "[SCHEME]", build_directory: "./DerivedData", output_directory: "./sonar-reports", proj: "./[PROJECT].xcodeproj")
    sh("cd .. && lizard ./[SOURCE_FOLDER] -l swift --xml > ./sonar-reports/lizard-report.xml")
    swiftlint(output_file: "./sonar-reports/swiftlint.txt", ignore_exit_status: true)
    sh("cd .. && mv sonar-reports/cobertura.xml sonar-reports/coverage-swift.xml")
    sonar
end
````

**Comment by qoomon** [comment]
just use the develop branch version of [run-sonar-swift.sh](https://github.com/Backelite/sonar-swift/blob/develop/sonar-swift-plugin/src/main/shell/run-sonar-swift.sh)

**Comment by Sicness** [comment]
The latest run-sonar-swift.sh doesn't help me:

```
$  ls sonar-reports                                                                                                                                                                                     
Prj-swiftlint.txt Prj-tailor.txt    TEST-report.xml       coverage-swift.xml
lizard-report.xml     xcodebuild.log
```
And anyway I got 
```INFO: Sensor org.sonar.plugins.swift.coverage.SwiftCoberturaSensor [backelitesonarswiftplugin]
INFO: Processing coverage report /Users/user/ios-prj/sonar-reports/coverage-swift.xml
WARN: Coverage measure for metric 'lines_to_cover' should not be saved directly by a Sensor. Plugin should be updated to use SensorContext::newCoverage instead.
WARN: Coverage measure for metric 'uncovered_lines' should not be saved directly by a Sensor. Plugin should be updated to use SensorContext::newCoverage instead.
WARN: Coverage measure for metric 'coverage_line_hits_data' should not be saved directly by a Sensor. Plugin should be updated to use SensorContext::newCoverage instead.
INFO: Sensor org.sonar.plugins.swift.coverage.SwiftCoberturaSensor [backelitesonarswiftplugin] (done) | time=196ms
INFO: Sensor org.sonar.plugins.swift.complexity.LizardSensor [backelitesonarswiftplugin]```

**Comment by Sicness** [comment]
Seems these warnings 
```
WARN: Coverage measure for metric 'lines_to_cover' should not be saved directly by a Sensor. Plugin should be updated to use SensorContext::newCoverage instead.
WARN: Coverage measure for metric 'uncovered_lines' should not be saved directly by a Sensor. Plugin should be updated to use SensorContext::newCoverage instead.
WARN: Coverage measure for metric 'coverage_line_hits_data' should not be saved directly by a Sensor. Plugin should be updated to use SensorContext::newCoverage instead. 
```

mean the report can't be found in path specified in **sonar.swift.coverage.reportPattern**. That's it. The bad error message.

**Comment by gaelfoppolo** [comment]
Do you still face this issue with the current version of the plugin?
Can you provided a project sample?

**Comment by gaelfoppolo** [comment]
Closing this issue. Please open a new one if you're experiencing this.
